### PR TITLE
 add selector parameter on ListRevisions method

### PR DESCRIPTION
**Pull request by gorilla001**
**What this PR does / why we need it**:
modified **ListRevisions** method

**Which issue(s) this PR fixes** 
no issue

**Special notes for your reviewer**:
none
**Release note**:
none
```release-note
none
```

### CODE REVIEW

File: pkg/controller/statefulset/stateful_set.go [hunk 1]
```diff
@@ -432,17 +432,17 @@ func (ssc *StatefulSetController) sync(key string) error {
...
 	if err != nil {
 		utilruntime.HandleError(fmt.Errorf("error converting StatefulSet %v selector: %v", key, err))
 		// This is a non-transient error, so don't retry.
 		return nil
 	}
 
-	if err := ssc.adoptOrphanRevisions(set); err != nil {
-		return err
-	}
-
```
Comment (krmayankk) [inline pkg/controller/statefulset/stateful_set.go]: having second thoughts now, if the selector  cannot be formed without err, what is the point of adopting orphan revisions, so we should leave the order

**Comment by krmayankk** [comment]
@kow3ns @janetkuo

File: pkg/controller/statefulset/stateful_set.go (hunk 1 above)
Comment (gorilla001) [reply pkg/controller/statefulset/stateful_set.go]: yes, you are right.but the selector seems to have been checked twice. once in adoptOrphanRevisions and another in here.

Comment (gorilla001) [reply pkg/controller/statefulset/stateful_set.go]: i'll make another change.

File: pkg/controller/statefulset/stateful_set.go [hunk 2]
```diff
@@ -313,18 +313,20 @@ func (ssc *StatefulSetController) getPodsForStatefulSet(set *apps.StatefulSet, s
 }
 
 // adoptOrphanRevisions adopts any orphaned ControllerRevisions matched by set's Selector.
-func (ssc *StatefulSetController) adoptOrphanRevisions(set *apps.StatefulSet) error {
-	revisions, err := ssc.control.ListRevisions(set)
+func (ssc *StatefulSetController) adoptOrphanRevisions(set *apps.StatefulSet, selector labels.Selector) error {
+	revisions, err := ssc.controllerHistory.ListControllerRevisions(set, selector)
```
Comment (janetkuo) [inline pkg/controller/statefulset/stateful_set.go]: This won't compile. Would you fix it?

Comment (gorilla001) [reply pkg/controller/statefulset/stateful_set.go]: Oh sorry, I'll fix it.

**Pull request closed without merging**